### ISSUE TITLE
Captcha image boxes cut off

### ISSUE THREAD

**Issue by shoff8** [clifford-fra/GoogleRecaptchav2#5]
The components installed correctly and I was able to place the Captcha box on my screen flow.  However, when using the flow, after I check the box and am presented with the image challenge, I can only see half of the bottom row of images.  Everything else is cut off.  No matter where on the screen I place the Captcha box this issue persists.  Any suggestions?

**Comment by gigaguy** [comment]
@shoff8 Did you get this fixed?  It works fine in debug mode, just not on the community page.  I tried adding every url I could find in 'Origin Page URL (optional)'

**Comment by Kwon25** [comment]
Same here if the origin page URL is set correctly as per the instructions.

**Comment by Kwon25** [comment]
Any updates please? :(

**Comment by pgroettjord** [comment]
Same here. Origin Page URL has been set. Flow appears fine in a published experience page. When clicking I'm not a robot, the challenge appears, but you have to scroll down to see all the images in the challenge. If scrolling to the bottom of the challenge to see the bottom images and clicking Confirm (without clicking on any images), it actually expands to show all the images.

**Comment by Kwon25** [comment]
I found work-around which is not great, but does the trick. It is to change the size of iframe's box with CSS on page load. 
Experience Builder > Theme > Edit CSS, add the following:

iframe.cGoogleRecaptcha {
    height: 600px;
    width: 500px;
}

Hope it helps.

**Comment by pgroettjord** [comment]
> iframe.cGoogleRecaptcha {
> height: 600px;
> width: 500px;
> }

That works great! Many thanks for sharing @Kwon25.

**Comment by Ashpork** [comment]
Although I'm working on a fork of this (to get some server side validation from @benedwards44 https://github.com/benedwards44/GoogleRecaptchav2) , I put a change into my "googleRecaptcha.cmp" (lines 20 and 24) to set the height at source to avoid any CSS troubles down the line.

`<iframe aura:id="captchaFrame" src="{!$Resource.Google_Recaptcha + '?sitekey=' + v.siteKey + '&#38;title=' + v.frameTitle}" width="350" height="550" style="border:0px; overflow:hidden;" />`

**Comment by Kwon25** [comment]
@Ashpork I guess this change is not yet on the repo here and we have to do it manually in our orgs is that correct? 

In regards to the server side validation - is it not doing currently server side validation?

**Comment by Ashpork** [comment]
Correct, yes I've done it locally, although more looking at the code today I should really spend a moment to see why the old logic didn't work any more. Currently my suggestion above causes the page to have a 550px height gap where the capture is, until you're proven to be human, then the page collapses back down to the right size. So really, the problem is that the page isn't expanding like it used to. I'm no .js developer, but might be able to unpick it. If I'm able to, I'll put a pull request in.

For server side validation, this specific repo is not doing it at the moment. see #4, BenEdwards44 has a pending merge request (#6 ) for the owner of this repo which contains it. But to be realistic, it's been waiting since Feb '22...

**Comment by Ashpork** [comment]
And too be fair, the recommendation from Salesforce will be to use what they've documented (https://www.learnexperiencecloud.com/article/Implementing-reCAPTCHA-in-Community-Cloud#ServerSideValidation). perhaps it's why this repo has gone quiet...

**Comment by Kwon25** [comment]
@Ashpork Thank you for that. I have seen that article but couldn't understand how it interacts with the screen flow to prevent the bot filling in the form and executing Next...perhaps I am not technical enough...but thank you for your help.
